### PR TITLE
chore(deps): upgrading `@readme/syntax-highlighter`

### PR DESCRIPTION
**Pull request by erunion**
### 🧰  Changes

Been a while since the syntax highlighter has been updated here and since I added support for TOML to it today, we should get it up to date here so Markdown code blocks with that language should get highlighted.

#### 10.2.0 (2020-10-29)

* feat: adding support for TOML highlighting (#26) ([b2e617b](https://github.com/readmeio/syntax-highlighter/commit/b2e617b)), closes [#26](https://github.com/readmeio/syntax-highlighter/issues/26)

#### <small>10.1.2 (2020-10-26)</small>

* fix: isolating regex check to highlightmode option (#25) ([aec7c81](https://github.com/readmeio/syntax-highlighter/commit/aec7c81)), closes [#25](https://github.com/readmeio/syntax-highlighter/issues/25)

#### <small>10.1.1 (2020-10-23)</small>

* fix: check for \n in CodeMirror runmode (#23) ([66444f4](https://github.com/readmeio/syntax-highlighter/commit/66444f4)), closes [#23](https://github.com/readmeio/syntax-highlighter/issues/23)

#### 10.1.0 (2020-10-22)

* chore: moving react and react-dom into being peerdeps (#24) ([d1826d7](https://github.com/readmeio/syntax-highlighter/commit/d1826d7)), closes [#24](https://github.com/readmeio/syntax-highlighter/issues/24)

#### <small>10.0.1 (2020-10-12)</small>

* feat(context): exporting context provider (#22) ([1dc67c0](https://github.com/readmeio/syntax-highlighter/commit/1dc67c0)), closes [#22](https://github.com/readmeio/syntax-highlighter/issues/22)
* fix(readonly): wip styled output (#21) ([82b99ab](https://github.com/readmeio/syntax-highlighter/commit/82b99ab)), closes [#21](https://github.com/readmeio/syntax-highlighter/issues/21)

#### 10.0.0 (2020-10-05)

* fix(bundle): client and server compatibility (#20) ([bd4362b](https://github.com/readmeio/syntax-highlighter/commit/bd4362b)), closes [#20](https://github.com/readmeio/syntax-highlighter/issues/20)
* chore(deps-dev): bump @commitlint/cli from 9.1.2 to 11.0.0 (#10) ([c466e8d](https://github.com/readmeio/syntax-highlighter/commit/c466e8d)), closes [#10](https://github.com/readmeio/syntax-highlighter/issues/10)
* chore(deps-dev): bump @commitlint/config-conventional (#19) ([b700798](https://github.com/readmeio/syntax-highlighter/commit/b700798)), closes [#19](https://github.com/readmeio/syntax-highlighter/issues/19)
* chore(deps-dev): bump css-loader from 3.6.0 to 4.3.0 (#16) ([e3c0c67](https://github.com/readmeio/syntax-highlighter/commit/e3c0c67)), closes [#16](https://github.com/readmeio/syntax-highlighter/issues/16)
* chore(deps-dev): bump eslint from 7.8.0 to 7.10.0 (#11) ([d5e30e0](https://github.com/readmeio/syntax-highlighter/commit/d5e30e0)), closes [#11](https://github.com/readmeio/syntax-highlighter/issues/11)
* chore(deps-dev): bump husky from 4.2.5 to 4.3.0 (#18) ([7e437fe](https://github.com/readmeio/syntax-highlighter/commit/7e437fe)), closes [#18](https://github.com/readmeio/syntax-highlighter/issues/18)
* chore(deps-dev): bump prettier from 2.1.1 to 2.1.2 (#14) ([f3c572f](https://github.com/readmeio/syntax-highlighter/commit/f3c572f)), closes [#14](https://github.com/readmeio/syntax-highlighter/issues/14)
* chore(deps-dev): bump sass-loader from 10.0.1 to 10.0.2 (#12) ([2e9a2e9](https://github.com/readmeio/syntax-highlighter/commit/2e9a2e9)), closes [#12](https://github.com/readmeio/syntax-highlighter/issues/12)
* chore(deps-dev): bump terser-webpack-plugin from 4.1.0 to 4.2.2 (#15) ([9c0d3dc](https://github.com/readmeio/syntax-highlighter/commit/9c0d3dc)), closes [#15](https://github.com/readmeio/syntax-highlighter/issues/15)
* chore(deps-dev): bump webpack from 4.44.1 to 4.44.2 (#17) ([828adca](https://github.com/readmeio/syntax-highlighter/commit/828adca)), closes [#17](https://github.com/readmeio/syntax-highlighter/issues/17)
* chore(deps): bump actions/checkout from v2.3.2 to v2.3.3 (#9) ([6f1fda0](https://github.com/readmeio/syntax-highlighter/commit/6f1fda0)), closes [#9](https://github.com/readmeio/syntax-highlighter/issues/9)
* chore(deps): bump codemirror from 5.57.0 to 5.58.1 (#13) ([7d7c7b3](https://github.com/readmeio/syntax-highlighter/commit/7d7c7b3)), closes [#13](https://github.com/readmeio/syntax-highlighter/issues/13)

#### <small>9.0.1 (2020-09-16)</small>

* chore: line number dom elements updated to use span (#8) ([85aeb0e](https://github.com/readmeio/syntax-highlighter/commit/85aeb0e)), closes [#8](https://github.com/readmeio/syntax-highlighter/issues/8)
* fix(highlighting): fixing an issue where overlay was not applied correctly (#7) ([df2236e](https://github.com/readmeio/syntax-highlighter/commit/df2236e)), closes [#7](https://github.com/readmeio/syntax-highlighter/issues/7)

#### 9.0.0 (2020-09-15)

* feat: code editor export (#5) ([3c2baa0](https://github.com/readmeio/syntax-highlighter/commit/3c2baa0)), closes [#5](https://github.com/readmeio/syntax-highlighter/issues/5) [#6](https://github.com/readmeio/syntax-highlighter/issues/6)
* docs: adding a line break in the readme ([6819099](https://github.com/readmeio/syntax-highlighter/commit/6819099))

#### <small>8.0.3 (2020-09-02)</small>

* chore(deps): upgrading @readme/variable to 7.2.1 ([7ec28f3](https://github.com/readmeio/syntax-highlighter/commit/7ec28f3))

#### <small>8.0.2 (2020-09-02)</small>

* docs: updating a link in the contributing docs ([c539940](https://github.com/readmeio/syntax-highlighter/commit/c539940))

#### <small>8.0.1 (2020-09-01)</small>

* feat: setting up webpack (#4) ([fa651fc](https://github.com/readmeio/syntax-highlighter/commit/fa651fc)), closes [#4](https://github.com/readmeio/syntax-highlighter/issues/4)

#### 8.0.0 (2020-08-31)

> No breaking changes in this release, the package has just moved a new home at https://github.com/readmeio/syntax-highlighter!

* chore: moving the package code into a src/ directory ([e78554e](https://github.com/readmeio/syntax-highlighter/commit/e78554e))
* chore: pulling the syntax-highlighter package over from the explorer repo ([296c7de](https://github.com/readmeio/syntax-highlighter/commit/296c7de))
* chore(deps-dev): bump eslint from 7.7.0 to 7.8.0 (#2) ([9789985](https://github.com/readmeio/syntax-highlighter/commit/9789985)), closes [#2](https://github.com/readmeio/syntax-highlighter/issues/2)
* chore(deps-dev): bump jest from 26.1.0 to 26.4.2 (#3) ([c837dd7](https://github.com/readmeio/syntax-highlighter/commit/c837dd7)), closes [#3](https://github.com/readmeio/syntax-highlighter/issues/3)
* chore(deps-dev): bump prettier from 2.0.5 to 2.1.1 (#1) ([0cfbe94](https://github.com/readmeio/syntax-highlighter/commit/0cfbe94)), closes [#1](https://github.com/readmeio/syntax-highlighter/issues/1)
* docs: setting up changelog generation ([dcfe1db](https://github.com/readmeio/syntax-highlighter/commit/dcfe1db))
* ci: adding a ci workflow for running tests ([fe631bd](https://github.com/readmeio/syntax-highlighter/commit/fe631bd))
* ci: setting up dependabot ([a293e53](https://github.com/readmeio/syntax-highlighter/commit/a293e53))
* test: fixing broken tests ([6257159](https://github.com/readmeio/syntax-highlighter/commit/6257159))

### CODE REVIEW

File: components/Code/index.jsx [hunk 1]
```diff
@@ -4,8 +4,8 @@ const PropTypes = require('prop-types');
 // Only load CodeMirror in the browser, for SSR
 // apps. Necessary because of people like this:
 // https://github.com/codemirror/CodeMirror/issues/3701#issuecomment-164904534
-const syntaxHighlighter = typeof window !== 'undefined' ? require('@readme/syntax-highlighter') : false;
-const canonicalLanguage = require('@readme/syntax-highlighter/canonical');
+const syntaxHighlighter = typeof window !== 'undefined' ? require('@readme/syntax-highlighter').default : false;
```
Comment (gratcliff) [inline components/Code/index.jsx:7]: Now that the server-compatible version of this,  `index.node.js` is the `main` entrypoint, you might need to reference the client version.

Here's `main`: https://github.com/readmeio/syntax-highlighter/blob/master/src/index.node.js (no default).
Here's the client: https://github.com/readmeio/syntax-highlighter/blob/master/src/index.js (default)

**Review by rafegoldberg — APPROVED** [review]
Thanks @erunion!

File: components/Code/index.jsx (hunk 1 above)
Comment (erunion) [reply components/Code/index.jsx:7]: Not sure what I need to change here.

Comment (erunion) [reply components/Code/index.jsx:7]: @kellyjosephprice You mind taking this? I'm not sure what exactly this require needs to be changed to.

Comment (kellyjosephprice) [reply components/Code/index.jsx:7]: Sure!

Comment (kellyjosephprice) [reply components/Code/index.jsx:7]: I believe there's no change needed? It looks like syntax-highlighter is using the `browser` field in `package.json` and the webpack build for markdown is loading the correct file.

```sh
rg SyntaxHighlighter dist
dist/main.node.js
5057:  }), console.log("SyntaxHighlighter loading server");

dist/main.js
20852:  console.log("SyntaxHighlighter loading client");
```

**Review by rafegoldberg — CHANGES_REQUESTED** [review]
Pulling back here as I just checked this out to verify, and it looks like there _are_ some SSR issues per Gabe's concerns. (To be fair though, I'm using `npm link` locally so, as always, I'm a touch skeptical as to how well this replicates a "true" install environment...)
```
[0] webpack:///./node_modules/codemirror/lib/codemirror.js?:17
[0]   var userAgent = navigator.userAgent;
```
<details><summary><b>Full Trace</b></summary>

```
[0] ReferenceError: navigator is not defined
[0]     at eval (webpack:///codemirror/lib/codemirror.js?:17:19)
[0]     at Object../node_modules/codemirror/lib/codemirror.js (/markdown/dist/main.node.js:510:1)
[0]     at __webpack_require__ (/Users/rafegoldberg/Repos/ReadMe/markdown/dist/main.node.js:21:30)
[0]     at eval (webpack:///./node_modules/@readme/syntax-highlighter/codemirror.jsx?:13:18)
[0]     at Object../node_modules/@readme/syntax-highlighter/codemirror.jsx (/Users/rafegoldberg/Repos/ReadMe/markdown/dist/main.node.js:310:1)
[0]     at __webpack_require__ (/Users/rafegoldberg/Repos/ReadMe/markdown/dist/main.node.js:21:30)
[0]     at eval (webpack:///./node_modules/@readme/syntax-highlighter/index.js?:3:18)
[0]     at Object../node_modules/@readme/syntax-highlighter/index.js (/Users/rafegoldberg/Repos/ReadMe/markdown/dist/main.node.js:321:1)
[0]     at __webpack_require__ (/Users/rafegoldberg/Repos/ReadMe/markdown/dist/main.node.js:21:30)
[0]     at eval (webpack:///./components/Code/index.jsx?:12:16)
[0]     at Object../components/Code/index.jsx (/Users/rafegoldberg/Repos/ReadMe/markdown/dist/main.node.js:119:1)
[0]     at __webpack_require__ (/Users/rafegoldberg/Repos/ReadMe/markdown/dist/main.node.js:21:30)
[0]     at eval (webpack:///./components/index.js?:5:63)
[0]     at Module../components/index.js (/Users/rafegoldberg/Repos/ReadMe/markdown/dist/main.node.js:230:1)
```

</details>

I can look a bit closer, unless you want to take this one @kellyjosephprice?

**Comment by kellyjosephprice** [comment]
Yea, looking now.

**Comment by rafegoldberg** [comment]
```diff
diff --git a/components/Code/index.jsx b/components/Code/index.jsx
index c66d2cec..e4ce96be 100644
--- a/components/Code/index.jsx
+++ b/components/Code/index.jsx
@@ -43,7 +43,9 @@ function Code(props) {
         suppressHydrationWarning={true}
       >
         {copyButtons && <CopyCode className="fa" code={children[0]} />}
-        {syntaxHighlighter ? syntaxHighlighter(children[0], language, { tokenizeVariables: true }) : children[0]}
+        {lang && syntaxHighlighter
+          ? syntaxHighlighter(children[0], language, { tokenizeVariables: true })
+          : children[0]}
       </code>
     </React.Fragment>
   );

```

**Comment by rafegoldberg** [comment]
So actually, I may have been on an outdated branch when I got [that error](https://github.com/readmeio/markdown/pull/61#pullrequestreview-525570900) ‘cause when I run this against next it's working. It _does_ look like these highlighter changes caused some minor problems rendering inline code snippets, though. I think we can fix this by adjusting [this line](https://github.com/readmeio/markdown/blob/576c821913538d31f3d4b236a7a4db0de3f5cde6/components/Code/index.jsx#L46) so that we only highlight a code block if it's `lang` value is explicitly defined?

```js
{lang && syntaxHighlighter
  ? syntaxHighlighter(children[0], language, { tokenizeVariables: true })
  : children[0]}
```

**Review by kellyjosephprice — APPROVED** [review]
Changes LGTM

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.22.1</b>](/readmeio/markdown/releases/tag/v6.22.1)
<br>

**Pull request merged**